pkg/api/pod/util.go

```diff
@@ -413,6 +423,16 @@ func dropDisabledVolumeDevicesFields(podSpec, oldPodSpec *api.PodSpec) {
 	}
 }
 
+// dropDisabledCSIVolumeSourceAlphaFields removes disabled alpha fields from []CSIVolumeSource.
+// This should be called from PrepareForCreate/PrepareForUpdate for all pod specs resources containing a CSIVolumeSource
+func dropDisabledCSIVolumeSourceAlphaFields(podSpec, oldPodSpec *api.PodSpec) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume) && !csiInUse(oldPodSpec) {
```

Comment:
&& csiInUse(oldPodSpec), remove the negation ?

Reply:
Fields are muted when the feature is disabled and the oldspec did not have csi in it. That's why its negated.